lib/elixir/lib/map.ex

```diff
@@ -691,8 +691,12 @@ defmodule Map do
       end
 
     case fun.(current) do
-      {get, update} -> {get, :maps.put(key, update, map)}
-      :pop          -> {current, :maps.remove(key, map)}
+      {get, update} ->
+        {get, :maps.put(key, update, map)}
+      :pop ->
+        {current, :maps.remove(key, map)}
+      other ->
+        raise "must return a two-element tuple or :pop, got: #{inspect(other)}"
```

Comment:
Should we raise an `ArgumentError` here? The erroneous argument would be `fun`, which returns a bad value thus being a bad argument. I think it makes sense? 😄 @josevalim wdyt?